src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -100,6 +110,9 @@ public void save(AddressBook addressBook) throws StorageOperationException {
         try (final Writer fileWriter =
                      new BufferedWriter(new FileWriter(path.toFile()))) {
 
+            if(!path.toFile().canWrite()){
```

Comment:
No need to check for the writability explicitly. Exception already gives you that information. 
All you need to do is to handle it!